reconfiguration/src/reconfiguration_handler.cpp

```diff
@@ -52,18 +53,26 @@ bool ReconfigurationHandler::handle(const KeyExchangeCommand& command,
                                     uint64_t sequence_number,
                                     uint32_t,
                                     const std::optional<bftEngine::Timestamp>&,
-                                    concord::messages::ReconfigurationResponse&) {
+                                    concord::messages::ReconfigurationResponse& rres) {
+  if (command.tls && command.target_replicas.size() > bftEngine::ReplicaConfig::instance().fVal) {
```

Comment:
what happens when we have only (n-f) functional nodes? should we ignore replica tls key exchange?

Reply:
There are two aspects here:
(1) whether to run the command at al - in this case, this is the user responsibility to update the clients correctly (or wait for the f faulty node to return)
(2) How the late replicas are catching up - this will be done by the state transfer reconfiguration mechanism
as for (1), the point is that replicas cannot know (at least without very complicated BFT failure detectors) that other replicas are not available
They can use some heuristics, but I prefer to leave it for the user's decision (at least for the fears phase). In the future, we can implement a timed exchanged mechanism (in which the replica runs the actual exchange after a predefined amount of time). But notice that even this mechanism will not solve the problem of n-f functional nodes if the client wasn’t able to get the update in the predefined timeout